### ISSUE TITLE
Android crashes around getPaymentSessionData

### ISSUE THREAD

**Issue by luke-hawk** [triniwiz/nativescript-stripe#118]
Hi @RobertGardner & @triniwiz. First, let me thank you for your work in maintaining this plugin!👍
However, I'm facing a strange bug in the **standard implementation** that I can't seem to fully trace: 

```ts
// stripe.service.ts
requestPayment(paymentSession: StripePaymentSession) {
    console.log(paymentSession);
    paymentSession.requestPayment();
}
// CONSOLE.LOG
// { customerSession: [Getter/Setter],
//    currency: [Getter/Setter],
//    listener: [Getter/Setter],
//    native: [Getter/Setter],
//    loading: [Getter/Setter],
//    receiver: [Getter/Setter],
//    [__ob__]:
//    { value: [Circular],
//        dep: { id: 2471, subs: [Object] },
//        vmCount: 0 },
//        selectedPaymentMethod: undefined,
//        selectedShippingMethod: undefined,
//        shippingAddress: undefined }
``` 
When further tracking down the problem my emulator seems to crash at `getPaymentSessionData`
```js
//standard.android.js
StripePaymentSession.prototype.requestPayment = function () {
  console.log('requestPayment')
  var _this = this;
  this.paymentInProgress = true;
  var data = this.native.getPaymentSessionData();
  console.log(data)
  ...
// CONSOLE.LOG
// 'requestPayment'

```
Strangely enough I don't get any Error Log / StackTrace with the crash – the emulator just shuts down. On iOS everything works perfectly fine so I suspect that my `stripe.service.ts` and other settings are correct. Oh, and as I'm not selling any physical good, I'm not requiring any shipping if that information helps...

Any idea? Any help highly appreciated and if there's anything I can do to help you, let me know!

EDIT: Not sure why `selectedPaymentMethod`  is undefined in the first place. May that be the cause of the issue? I've definitely selected a card though (backend currently is the stripe-demo on heroku)  

EDIT 2: Logging `onPaymentDataChanged(data)` in the Listener also yields `paymentMethod: undefined`

**Comment by RobertGardner** [comment]
Strange crash. I've also been having occasional Android crashes when one of the expected fields is missing. Same behavior -- the emulator just crashes. There may be an issue if there is no shipping data, but it looks like the plugin code is correct. (Ha! Famous last words!)

In the meantime, I've received a PR (#117) from @jlbuganan to update the Android Stripe SDK to v12.7. Let me work on that today, then you can try that version and see if, by some miracle, that fixes the problem. If not, I'll try modifying the demo to add a scenario with no shipping info and see if I can reproduce the crash.

**Comment by luke-hawk** [comment]
@RobertGardner Yeah I can't really get to the root of this bug. Really strange. Let me know when your done with updating to Android SDK v12.7 and I'll be glad to give it another try! Thanks for your help – really appreciate it! :-)

**Comment by RobertGardner** [comment]
I couldn't upgrade to Stripe Android SDK v12.7 because of an apparent {NS} bug with Kotlin libraries. But, I created a demo in `demo-angular` that does not require shipping. It works successfully on both iOS and Android. It's submitted if you want to see it.

One thing that comes to mind is the new NS Android marking mode. Do you have `"markingMode": "none"` in your `package.json`? I'm not convinced that the Stripe SDK works correctly with marking mode `none` since it does a lot of work under the covers. I've tried `none` on my own app a bit, but not extensively, and haven't (yet) run into problems, but just reading the documentation worries me. I'm not sure what will happen when `none` becomes the only way to run! On the other hand, I would expect crashes to be intermittent if this is the problem. If you get them consistently then I may be barking up the wrong tree.

Other than that, I'm not sure what the problem could be. Have you tried debugging and stepping into the plugin code? Do you know which Android call is being made when it crashes?

**Comment by luke-hawk** [comment]
@RobertGardner Thanks a lot for your help! I finally found the root for this issue: Apparently the page onload event fired whenever the Stripe payment method modal closed and therefore initiated a new payment session. Note: I'm on nativescript-vue and used PR #84 as a demo. The issue is quite easy to resolve though (just had hard times actually finding the cause) – just initiate the payment session from vue's `created()` hook instead of the `@loaded` event of the page. Let me know if I can help with updating the vue demo and/or the docs. I've also been able to use the standard implementation with intents which actually might be quite a common use case for the  europeans among us. 

Oh and I've tried both `"markingMode": "none"` and `"markingMode": "full"` and couldn't tell a difference in my case!
